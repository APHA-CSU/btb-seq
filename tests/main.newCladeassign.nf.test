nextflow_process {

    name "Test Process NEWCLADEASSIGN"
    script "modules/newCladeassign.nf"
    process "NEWCLADEASSIGN"

    test("Should assign clade") {

        when {
            params {
                csstable = "${projectDir}/references/CSSwithref.csv"
            }
            process {
                """
                input[0] = ["phix-m", file("${projectDir}/tests/data/phix_test/combine_tests/phix-m_S800_consensus.fas"),
                            file("${projectDir}/tests/data/phix_test/combine_tests/phix-m_S800_snps.tab")]
		        input[1] = params.csstable
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
