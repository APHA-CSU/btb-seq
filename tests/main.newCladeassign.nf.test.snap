{
    "Should assign clade": {
        "content": [
            {
                "0": [
                    "phix-m_cladematch.csv:md5,7dd2f5708232977848a53d355a481203"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-14T13:05:04.136809215"
    }
}